src/library/Query.js

```diff
@@ -33,11 +45,49 @@ function query(httpMethod, deployment, apiKey, qry, payload) {
 }
 
 function limitQuery(httpMethod, deployment, apiKey, queryPath, limit, offset, payload, address) {
-  // validate and normalize limit and offset
-  const queryOptions = buildQueryOptions(limit, offset, address);
+  const separationLimit = 30;
+  const limitChecked = checkLimit(limit);
+  if (limitChecked > separationLimit) {
+    const results = {
+      status: null,
+      message: null,
+    };
+    // MBQUERY, MBCUSTOMQUERY has rows in result but MBEVENTS has not
+    const hasRows = !/^events$/.test(queryPath);
+    const rows = [];
 
-  // query
-  const results = query(httpMethod, deployment, apiKey, queryPath + queryOptions, payload);
+    let offsetNext = offset;
```

Comment:
if `offset` wasn't provided (`offset == undefined`), then `offsetNext == offset == undefined`, and lower down, `offsetNext += separationLimit == NaN`.

![image](https://user-images.githubusercontent.com/5861379/101489299-1aae9180-39a4-11eb-89de-947e25cb9bac.png)

In the spreadsheet:

![image](https://user-images.githubusercontent.com/5861379/101489827-d8d21b00-39a4-11eb-8de5-c035eff4c805.png)

Reply:
We should also add a test to check this case.